GTViewController/Classes/Nav Classes/GTViewController.m

```diff
@@ -322,16 +324,20 @@ - (void)registerListenerWithName:(NSString *)eventName forPageNumber:(NSInteger)
 }
 
 - (void)pageEventDetected:(NSNotification *)notification {
-	
-	NSString *eventName = notification.userInfo[GTPageNotificationEventKeyEventName];
-	
-	if (eventName && self.listeners[eventName]) {
-		
-		NSNumber *pageNumber = self.listeners[eventName][0];
-		[self switchToPageWithIndex:pageNumber.integerValue];
-		[self.centerPage triggerEventWithName:eventName];
-	}
-	
+    
+    NSString *eventName = notification.userInfo[GTPageNotificationEventKeyEventName];
+    
+    if (eventName && self.listeners[eventName]) {
+        
+        NSNumber *pageNumber = self.listeners[eventName][0];
+        [self switchToPageWithIndex:pageNumber.integerValue];
+        [self.centerPage triggerEventWithName:eventName];
+        
+        // dismiss a follow up modal if it is present and displayed
+        if (self.followupViews) {
```

Comment:
this works. It might be better to save the followupView with the event in self.listeners when it is registered in the registerListener method. It is more flexible that way